src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -331,6 +331,16 @@ private IEnumerable<CustomNodeInfo> ScanNodeHeadersInDirectory(string dir, bool
         /// </summary>
         private void SetNodeInfo(CustomNodeInfo newInfo)
         {
+            var guids = NodeInfos.Where(x =>
+                        {
+                            return string.Compare(x.Value.Path, newInfo.Path, StringComparison.OrdinalIgnoreCase) == 0;
+                        }).Select(x => x.Key).ToList();
+
+            foreach (var guid in guids)
```

Comment:
If we do not convert it to list, when iterating through guids which depend on NodeInfos, it does not allow changes to NodeInfos or it will cause an error.